src/explore-education-statistics-common/src/modules/charts/util/getUnit.ts

```diff
@@ -0,0 +1,23 @@
+import { DataSetCategoryConfig } from '@common/modules/charts/util/getDataSetCategoryConfigs';
+
+const getUnit = (
+  configUnit: string | undefined,
+  dataSetCategoryConfigs: DataSetCategoryConfig[],
+) => {
+  if (configUnit) {
+    return configUnit;
+  }
+
+  const units = dataSetCategoryConfigs.map(config => {
+    return config.dataSet.indicator.unit;
+  });
+
+  // Check if all units are the same, leave blank if not.
+  // Don't display £ as unable to position them before the value. (EES-2278)
+  if (units.every(unit => unit === units[0])) {
```

Comment:
Could instead of writing:

```
if (units.every(unit => unit === units[0])) {
    return units[0] === '£' ? '' : units[0];
}

return '';
```

write instead:

```
return units[0] === '£' || !units.every(unit => unit === units[0])) ? '' : units[0];
```

I guess at this point we're happy that our `dataSetCategoryConfigs` array has at least one element?

Reply:
I've made this change now. 

This would have returned undefined before if `dataSetCategoryConfigs` was  empty. This is what was previously happening:

* If `dataSetCategoryConfigs` is empty then:
* `units` is empty
*  so `units[0]` is undefined
*  so `units[0] === '£'` evaluates to false
* so !units.every(unit => unit === units[0]) evaluates to false (since the condition is ignored, `every` returns true)
* so `units[0]` is returned which is undefined.

Now added an addition check `units.length === 0` to return `''` rather than undefined.